### PR TITLE
Improved response documentation structure and added encoding helper function

### PR DESCRIPTION
**Pull request by shamasis**

### CODE REVIEW

**Comment by shamasis** [comment]
@numaanashraf - using this, we can temporarily implement lib decompression in app response if 

```
var enc = response.encoding();

if (enc.format === 'gzip' && enc.source === 'body') {
     response.stream = (new zlib.gunzip()).pipe(response.stream); // pseudo-code
}
```

File: lib/collection/response.js [hunk 1]
```diff
@@ -63,7 +110,7 @@ _.assign(Response.prototype, /** @lends Response.prototype */ {
              * @type {String}
              * @deprecated use .reason()
              */
-            status: options.status ? options.status : httpReasons.lookup(options.code).name,
+            status: _.pluck(options.reason, options.status, httpReasons.lookup(options.code).name),
```
Comment (czardoz) [inline lib/collection/response.js]: `_.pluck` used to be a lodash 3 function which used to do something different. Sure we should use that name?

File: lib/collection/response.js [hunk 2]
```diff
@@ -246,6 +304,38 @@ _.assign(Response.prototype, /** @lends Response.prototype */ {
...
+     * @private
+     * @returns {Object} - {format: string, source: string}
+     */
+    encoding: function () {
+        var contentEncoding = Header.headerValue(this.headers, CONTENT_ENCODING),
+            body = this.stream || this.body,
+            source;
+
+        if (contentEncoding) {
+            source = 'header';
```
Comment (czardoz) [inline lib/collection/response.js]: Probably should be `HEADER`

File: lib/util.js [hunk 3]
```diff
@@ -184,6 +186,14 @@ _.mixin(/** @lends util */ {
             result[i] = ASCII_SOURCE[(Math.random() * ASCII_SOURCE_LENGTH) | 0];
         }
         return result.join(EMPTY);
+    },
+
+    pluck: function () {
```
Comment (czardoz) [inline lib/util.js]: https://github.com/postmanlabs/postman-collection/pull/192/files#r86092342

**Comment by czardoz** [comment]
@kunagpal, this is a simple change with not many comments, can you update the PR?

**Comment by kunagpal** [comment]
@czardoz Alright, I'll get on this.

File: lib/util.js (hunk 3 above)
Comment (kunagpal) [inline lib/util.js]: How about renaming to pick?

Comment (czardoz) [inline lib/util.js]: `_.choose` sounds closer to the original intended meaning

**Comment by kunagpal** [comment]
@czardoz Are there any other changes needed here?

**Pull request merged**